components/features/currency/amount-select/amount-select.tsx

```diff
@@ -1,77 +1,91 @@
 import { Select, SelectItem } from "@nextui-org/react";
+import { ChangeEvent } from "react";
 import { Money } from "utils/Money/Money";
 
 import { Chip } from "src/components/Chip/Chip";
 import { CurrencyIcons } from "src/components/Currency/CurrencyIcon";
+import { useCurrenciesOrder } from "src/hooks/useCurrenciesOrder";
 import { useIntl } from "src/hooks/useIntl";
 
 import { Input } from "components/ds/form/input/input";
-import { TAmoutSelect } from "components/features/currency/amount-select/amount-select.types";
-
-const items = [
-  {
-    label: "Dollars (USD)",
-    value: "USD",
-  },
-  {
-    label: "Aptos (APT)",
-    value: "APT",
-  },
-  {
-    label: "Optimism (OP)",
-    value: "OP",
-  },
-  {
-    label: "Ether (ETH)",
-    value: "ETH",
-  },
-  {
-    label: "Stark (STRK)",
-    value: "STRK",
-  },
-] as const;
+import { TAmountSelect } from "components/features/currency/amount-select/amount-select.types";
 
 // TODO handle blue style
-// TODO handle dynamic select items
-export function AmountSelect({ inputProps }: TAmoutSelect.Props) {
+export function AmountSelect({
+  inputProps,
+  budgets,
+  amountValue,
+  selectionValue,
+  onAmountChange,
+  onSelectionChange,
+}: TAmountSelect.Props) {
   const { T } = useIntl();
+  const orderedCurrencies = useCurrenciesOrder({ currencies: budgets });
+
+  const handleSelectionChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    const relatedBudgetCurrency = budgets?.find(budget => budget.currency.code === e.target.value)?.currency;
+    if (e.target.value && relatedBudgetCurrency) {
+      onSelectionChange?.(relatedBudgetCurrency);
+    }
+  };
+
+  const handleAmountChange = (amount: string) => {
+    onAmountChange?.(amount);
+  };
 
   return (
     <Input
-      type="number"
+      type="text"
       placeholder="0.00"
-      size={"lg"}
-      radius={"lg"}
+      size="lg"
+      radius="full"
+      className="h-11"
+      disabled={inputProps?.disabled || !orderedCurrencies?.length}
+      value={amountValue}
+      onChange={e => handleAmountChange(e.target.value.replace(/[^0-9.,]+/g, "").replace(/,/g, "."))}
       endContent={
-        <div className="flex w-[260px] items-center">
+        <div className="flex w-fit items-center">
           <Select
             aria-label={T("v2.commons.currency")}
-            startContent={
-              <Chip solid className="h-5 w-5 flex-shrink-0">
-                <CurrencyIcons currency={Money.fromSchema({ code: "USD" })} className="h-5 w-5" />
-              </Chip>
-            }
-            defaultSelectedKeys={["OP"]}
+            defaultSelectedKeys={[selectionValue?.code]}
+            selectedKeys={[selectionValue?.code]}
             classNames={{
-              trigger: "p-0 h-auto !bg-transparent shadow-none",
+              trigger: "p-0 h-auto !bg-transparent shadow-none flex flex-row items-center space-x-4",
```

Comment:
`flex` uses `flex-row` by default

```suggestion
              trigger: "p-0 h-auto !bg-transparent shadow-none flex items-center space-x-4",
```